### PR TITLE
Concurrent Merkle Tree: Improve Docs

### PR DESCRIPTION
**Pull request by ngundotra**
Address Halborn feedback on documentation & public function visibility

### CODE REVIEW

**Comment by jarry-xiao** [comment]
I hate clippy so much

File: libraries/concurrent-merkle-tree/src/concurrent_merkle_tree.rs [hunk 1]
```diff
@@ -81,9 +81,16 @@ impl<const MAX_DEPTH: usize, const MAX_BUFFER_SIZE: usize>
         Self::default()
     }
 
+    pub fn is_initialized(&self) -> bool {
```
Comment (jarry-xiao) [inline libraries/concurrent-merkle-tree/src/concurrent_merkle_tree.rs:84]: I think `sequence_number` is enough here, but this is fine

File: libraries/concurrent-merkle-tree/tests/tests.rs [hunk 2]
```diff
@@ -29,6 +29,13 @@ async fn test_initialize() {
         off_chain_tree.get_root(),
         "Init failed to set root properly"
     );
+
+    // Check that reinitialization fails
+    if let Err(ConcurrentMerkleTreeError::TreeAlreadyInitialized) = cmt.initialize() {
+        println!("Reinitialization successfully prevented");
+    } else {
+        assert!(false, "Tree should not be able to be reinitialized");
```
Comment (jarry-xiao) [inline libraries/concurrent-merkle-tree/tests/tests.rs:37]: `panic!` will appease the clippy gods

Comment (ngundotra) [reply libraries/concurrent-merkle-tree/tests/tests.rs]: thank you

**Comment by ngundotra** [comment]
Merged

**Pull request merged**